aten/src/ATen/test/cuda_vectorized_test.cu

```diff
@@ -0,0 +1,124 @@
+#include <gtest/gtest.h>
+#include <ATen/ATen.h>
+#include <ATen/native/cuda/MemoryAccess.cuh>
+#include <ATen/cuda/CUDAContext.h>
+
+using namespace at::native::memory;
+__managed__ double4 buffer1[1024];
+__managed__ double4 buffer2[1024];
+
+void reset_buffers() {
+  for (int i = 0; i < 1024; i++) {
+    buffer1[i].x = i;
+    buffer1[i].y = i + 0.1;
+    buffer1[i].z = i + 0.2;
+    buffer1[i].w = i + 0.3;
+
+    buffer2[2].x = -i;
+    buffer2[2].y = -(i + 0.1);
+    buffer2[2].z = -(i + 0.2);
+    buffer2[2].w = -(i + 0.3);
+  }
+}
+
+TEST(TestVectorizedMemoryAccess, CanVectorizeUpTo) {
+  char *ptr = reinterpret_cast<char *>(buffer1);
+  std::cout << ((uint64_t)ptr) << std::endl;
+
+  ASSERT_EQ(can_vectorize_up_to<bool>(ptr), 4);
+  ASSERT_EQ(can_vectorize_up_to<int8_t>(ptr), 4);
+  ASSERT_EQ(can_vectorize_up_to<int16_t>(ptr), 4);
+  ASSERT_EQ(can_vectorize_up_to<int>(ptr), 4);
+  ASSERT_EQ(can_vectorize_up_to<int64_t>(ptr), 4);
+
+  ASSERT_EQ(can_vectorize_up_to<bool>(ptr + 1), 1);
+  ASSERT_EQ(can_vectorize_up_to<int8_t>(ptr + 1), 1);
+
+  ASSERT_EQ(can_vectorize_up_to<bool>(ptr + 2), 2);
+  ASSERT_EQ(can_vectorize_up_to<int8_t>(ptr + 2), 2);
+  ASSERT_EQ(can_vectorize_up_to<int16_t>(ptr + 2), 1);
+
+  ASSERT_EQ(can_vectorize_up_to<bool>(ptr + 4), 4);
+  ASSERT_EQ(can_vectorize_up_to<int8_t>(ptr + 4), 4);
+  ASSERT_EQ(can_vectorize_up_to<int16_t>(ptr + 4), 2);
+  ASSERT_EQ(can_vectorize_up_to<int>(ptr + 4), 1);
+
+  ASSERT_EQ(can_vectorize_up_to<bool>(ptr + 8), 4);
+  ASSERT_EQ(can_vectorize_up_to<int8_t>(ptr + 8), 4);
+  ASSERT_EQ(can_vectorize_up_to<int16_t>(ptr + 8), 4);
+  ASSERT_EQ(can_vectorize_up_to<int>(ptr + 8), 2);
+  ASSERT_EQ(can_vectorize_up_to<int64_t>(ptr + 8), 1);
+}
+
+template <typename scalar_t, int vec_size>
+__global__ void vectorized_copy(scalar_t *dst, scalar_t *src) {
+  using vectorized = policies<64, 4>::vectorized<vec_size>;
+  auto policy = vectorized();
+  scalar_t buf[vectorized::thread_work_size];
+  auto accessor = [&](int index) -> scalar_t & { return buf[index]; };
+  policy.load(accessor, src + 256 * blockIdx.x);
+  policy.store(dst + 256 * blockIdx.x, accessor);
+}
+
+TEST(TestVectorizedMemoryAccess, CopyKernel) {
+  if (!at::cuda::is_available()) return;
```

Comment:
The internal linters want all if statements (and similar control-flow statements) to use braces:

`if (cond) {
...
}`

Reply:
fixed